src/ape/api/providers.py

```diff
@@ -268,69 +337,162 @@ class ProviderAPI:
 
     @abstractmethod
     def connect(self):
-        ...
+        """
+        Connect a contract to a provider.
+        """
 
     @abstractmethod
     def disconnect(self):
-        ...
+        """
+        Disconnect a contract from a provider.
+        """
 
     @abstractmethod
     def update_settings(self, new_settings: dict):
-        ...
+        """
+        Change the settings connection protocol.
+        May require a reconnect.
+
+        Agrs:
+            new_settings (dict): Value of new provider
+        """
 
     @property
     @abstractmethod
     def chain_id(self) -> int:
-        ...
+        """
+        Gives value of blockchain id.
+
+        Returns:
+            int: value of blockchain id.
+        """
 
     @abstractmethod
     def get_balance(self, address: str) -> int:
-        ...
+        """
+        Gives values of address balance.
+
+        Args:
+            address (str): Address string.
+
+        Returns:
+            int: value of balance at the address.
+        """
 
     @abstractmethod
     def get_code(self, address: str) -> bytes:
-        ...
+        """
+        Get the bytes of the contract.
+
+        Args:
+            address (str): value of address (Required)
+
+        Returns:
+        bytes (int): HexBytes of contract.
+        """
 
     @abstractmethod
     def get_nonce(self, address: str) -> int:
-        ...
+        """
+        Gets the number of transactions.
+
+        Agrs:
+            address (str): string of address.
+
+        Returns:
+            int: value of nonce.
+        """
 
     @abstractmethod
     def estimate_gas_cost(self, txn: TransactionAPI) -> int:
-        ...
+        """
+        Use for setting the default gas limit on a transaction.
+
+        Args:
+            txn (:class:`~ape.api.providers.TransactionAPI`) :
+            The transaction to estimate the gas for.
+
+        Returns:
+            int: value of estimated gas cost.
+        """
 
     @property
     @abstractmethod
     def gas_price(self) -> int:
-        ...
+        """
+        Value of gas.
+
+        Returns:
+            int
+        """
 
     @property
     def priority_fee(self) -> int:
+        """
+         A miner tip to incentivize them
+        to include your transaction in a block.
```

Comment:
```suggestion
        to include your transaction in a block.
        Only providers that implement EIP-1559 will have this property,
        otherwise will raise ``NotImplementedError``.
```